### DISCUSSION TITLE
[Inactivity] Condition to another entity

### DISCUSSION THREAD

**Discussion by txusebe** [ExperienceLovelace/ha-floorplan#12] [Q&A] [answered]
Hi everybody, i think it is a simple question if it would be possible.
I want to condition an elment to itself and to another entity too, but i dont know what is de correct sintax
I mean i need something like this

>     - entities:
>         - input_number.test1
>       tap_action: more-info
>       hover_action: hover-info
>       state_action:
>         action: call-service
>         service: floorplan.image_set
>         service_data: |
>           >
>           if (entity.state < 20){
>             if (input_number.test2 <20){
>               imageName = "image1
>            ...

but the problem is how to write correctly the second if expresion to reffer de test2

thank you

**Comment by exetico** [comment]
Hi

So, you could like to create a condition, and use x image, if X value from `input_number.test2` are below 20?

Or `if entity.state < 20` **AND** `if input_number.test2 < 20?  

If I understand correctly, it's a question using two conditions, and the same time. right?

If so, I'm catually not sure, how to use `input_number` in the same condition. I've not seen any examples from @pkozul  - So i guess we need to wait a bit.


${(entity.state === "on") ? (entity.state === "on") ? "light-on" : "light-off" : "light-off"}

`states` and `is_state` are not build in to our logics, so I guess this should be a new feature-request. I'll see if where's a way to do it "as is", but I doubt it. One solution would be to include an option to use other states - and another could be, to add multiple contitions.

**But** for your example, you could group entity.state and input_numer, if the values was `on`. I'm not sure how it's handled, now it's a numeric value.

**Comment by txusebe** [comment]
it is only a test to do my own example...  you must pay attention y have two inputs `test1` and `test2` and i need to check both to set several images `image1` `image2` `image3`... `imageN`

**Comment by exetico** [comment]
Could you please create a real-world example, with the names and entities you'll like to use? (but with english characters, ofc.)

Please wrap your YAML like this:
![image](https://user-images.githubusercontent.com/3549445/103559248-1f3c6c00-4eb6-11eb-9234-9f484cf693b0.png)

If I was in the same situation as you, I would create e template_sensor in HA, where I would just all the sensor data I could ever need, and define the `image1` value in the template_sensor. In that case, you could just use the template_sensor value in your rule... I guess? :-)

**Comment by exetico** [comment]
Do you still need more inspiration and help to this, or can I close the issue? It's more related to Home Assistant-usage in general, cause ha-floorplan will be too bloated, if we start handling things like this :-) (So it will only be considered, if we're seeing a high amount of requests related to this)

**Comment by exetico** [comment, marked as answer]
I'll mask this as the answer, due to inactivity.

**Reply by pkozul** [reply]
Hi @txusebe 

We still need to document this stuff, but here is some info from the earlier version of Floorplan, that still applies to the new version:

> The full set of objects available to your template literals is shown below:
> 
> `entity` - the state object for the current entity
> `entities` - the state objects for all entities
> `hass` - the hass object
> `config` - the floorplan configuration
> 

So, if you want to reference some other entities in your floorplan rule, you should be able to do that using the `entities` object. For example:

```
  if (entities["input_number.test2"].state < 20) {
```

Does that work for you?
